src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -864,54 +1063,101 @@ private static void VerifyRefAssemblyClient(string lib_cs, string source, Action
+        {
+            var id1 = ModuleMetadata.CreateFromImage(firstImage).GetMetadataReader().ReadAssemblyIdentityOrThrow();
+            var id2 = ModuleMetadata.CreateFromImage(secondImage).GetMetadataReader().ReadAssemblyIdentityOrThrow();
+            Assert.Equal(expectMatch, id1 == id2);
+        }
+
+        private (ImmutableArray<byte> image, ImmutableArray<byte> refImage) EmitRefOut(CSharpCompilation comp)
+        {
+            using (var output = new MemoryStream())
+            using (var metadataOutput = new MemoryStream())
             {
-                using (var output = new MemoryStream())
-                using (var metadataOutput = new MemoryStream())
-                {
-                    var options = EmitOptions.Default.WithIncludePrivateMembers(false);
-                    comp.VerifyEmitDiagnostics();
-                    var result = comp.Emit(output, metadataPEStream: metadataOutput,
-                        options: options);
-                    return (output.ToImmutable(), metadataOutput.ToImmutable());
-                }
+                var options = EmitOptions.Default.WithIncludePrivateMembers(false);
+                comp.VerifyEmitDiagnostics();
+                var result = comp.Emit(output, metadataPEStream: metadataOutput,
+                    options: options);
+                return (output.ToImmutable(), metadataOutput.ToImmutable());
             }
+        }
 
-            ImmutableArray<byte> emitRefOnly()
+        ImmutableArray<byte> EmitRefOnly(CSharpCompilation comp)
+        {
+            using (var output = new MemoryStream())
             {
-                using (var output = new MemoryStream())
-                {
-                    var options = EmitOptions.Default.WithEmitMetadataOnly(true).WithIncludePrivateMembers(false);
-                    comp.VerifyEmitDiagnostics();
-                    var result = comp.Emit(output,
-                        options: options);
-                    return output.ToImmutable();
-                }
+                var options = EmitOptions.Default.WithEmitMetadataOnly(true).WithIncludePrivateMembers(false);
+                comp.VerifyEmitDiagnostics();
+                var result = comp.Emit(output,
+                    options: options);
+                return output.ToImmutable();
             }
         }
 
+        [Fact]
+        public void RefAssembly_PublicSigning()
+        {
+            var snk = Temp.CreateFile().WriteAllBytes(TestResources.General.snKey);
+
+            var comp = CreateStandardCompilation("public class C{}",
+                options: TestOptions.ReleaseDll.WithCryptoKeyFile(snk.Path).WithPublicSign(true));
+
+            comp.VerifyDiagnostics();
+            var (image, refImage) = EmitRefOut(comp);
+            var refOnlyImage = EmitRefOnly(comp);
+            VerifyIdentitiesMatch(image, refImage);
```

Comment:
I did when debugging (public key shows in identity). I'll add check to the test.